src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb

```diff
@@ -173,7 +173,9 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
                 Dim iBaseAddress As UInteger
 
-                If (TypeOf (value) Is UInteger) Then
+                Dim throwAwayObject As Decimal = Nothing
+                If (TypeOf (value) Is UInteger OrElse
+                    (CpsPropertyDescriptorWrapper.IsAnyCpsComponent(m_Objects) AndAlso Decimal.TryParse(DirectCast(value, String), throwAwayObject))) Then
                     iBaseAddress = CUInt(value)
```

Comment:
I don't think there's a way today to get anything other than a string. There are other places in the prop pages where such code exists. However, for compat, it'll be good if the object returned here was the right data type instead of string.